impl/src/planning/subplanners.rs

```diff
@@ -335,6 +335,24 @@ impl<'planner> CrateSubplanner<'planner> {
     let is_workspace_member_dependency = !&workspace_member_dependents.is_empty();
     let is_binary_dependency = self.settings.binary_deps.contains_key(&package.name);
 
+    let mut raze_settings = self.crate_settings.cloned().unwrap_or_default();
+
+    // Generate canonicalized paths to additional build files so they're guaranteed to exist
+    // and always locatable.
+    if let Some(additional_build_file) = raze_settings.additional_build_file {
+      let canonicalized_path = cargo_workspace_root
+        .join(&additional_build_file)
+        .canonicalize()
+        .with_context(|| {
+          format!(
+            "Failed to find additional_build_file: {}",
+            &additional_build_file.display()
+          )
+        })?;
+
+      raze_settings.additional_build_file = Some(canonicalized_path);
```

Comment:
What do you think about adding a field to CrateContext "canonical_additional_build_file" instead of mutating the settings?

This seems like a minor quibble, but for some reason I'm still peeved about this. I think I view the settings as sacrosanct, and the mutation of the value here in the middle of the logic seems like heresy.

(at a meta level, I'm still trying to understand why this bothers me....)

Reply:
(then we can pull the canonical_additional_build_file right out and use it here: https://github.com/google/cargo-raze/blob/4d1721ed32e19dfea8794f868a4884bdffdc4014/impl/src/rendering/bazel.rs#L217)

Reply:
Sorry, just saw this suggestion but I think I updated this accordingly?